components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/basicauth/BasicAuthAuthenticator.java

```diff
@@ -108,19 +104,18 @@ public AuthenticationResponse authenticate(MessageContext synCtx) {
             log.info("Basic Authentication initialized");
         }
 
-        if (swagger == null && apiUUID != null) {
-            Entry localEntryObj = (Entry) synCtx.getConfiguration().getLocalRegistry().get(apiUUID);
-            if (localEntryObj != null) {
-                SwaggerParser parser = new SwaggerParser();
-                swagger = parser.parse(localEntryObj.getValue().toString());
-            }
+        openAPI = (OpenAPI) synCtx.getProperty(APIMgtGatewayConstants.API_SWAGGER);
+        if (openAPI == null) {
+            log.error("Swagger is missing in the gateway. Basic authentication cannot be performed.");
+            return new AuthenticationResponse(false, isMandatory, true,
+                    APISecurityConstants.API_AUTH_MISSING_SWAGGER, "Basic authentication cannot be performed.");
```

Comment:
Shall we use openAPI instead of Swagger?

Reply:
done